File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/LocationActionLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class LocationActionLabelsTest {
  @Test
  fun `labels availability action by current state`() {
    assertEquals("Comecar corridas", availabilityActionLabel(available = false, loading = false))
    assertEquals("Parar corridas", availabilityActionLabel(available = true, loading = false))
  }

  @Test
  fun `labels availability action while saving`() {
    assertEquals("Salvando...", availabilityActionLabel(available = true, loading = true))
  }

  @Test
  fun `labels automatic location action without duplicating pause wording`() {
    assertEquals("Ligar GPS ao vivo", automaticLocationActionLabel(trackingEnabled = false))
    assertEquals("Parar GPS ao vivo", automaticLocationActionLabel(trackingEnabled = true))
  }

  @Test
  fun `labels manual location action`() {
    assertEquals("Enviar minha posicao", sendLocationActionLabel(loading = false))
    assertEquals("Enviando...", sendLocationActionLabel(loading = true))
  }
}

File: apps/motoboy-pwa/src/locationActionLabels.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import {
  automaticLocationActionLabel,
  availabilityActionLabel,
  availabilityShortActionLabel,
  sendLocationActionLabel,
} from "./locationActionLabels";

test("labels availability action by current state", () => {
  assert.equal(availabilityActionLabel(false), "Comecar corridas");
  assert.equal(availabilityActionLabel(true), "Parar corridas");
  assert.equal(availabilityShortActionLabel(), "Corridas");
});

test("labels automatic location action clearly", () => {
  assert.equal(automaticLocationActionLabel(false), "Ligar GPS ao vivo");
  assert.equal(automaticLocationActionLabel(true), "Parar GPS ao vivo");
});

test("labels manual location action", () => {
  assert.equal(sendLocationActionLabel(false), "Enviar minha posicao");
  assert.equal(sendLocationActionLabel(true), "Enviando...");
});
